### PR TITLE
Nbdev migration

### PR DESCRIPTION
**Pull request by sjkoelle**
Remove nbdev from repo because its annoying.

### CODE REVIEW

**Comment by williamluer** [comment]
Do we need all 16 LJ files?

File: tests/tests/models/test_tacotron2.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        (
+            train_set,
+            val_set,
+            train_loader,
+            sampler,
+            collate_fn,
+        ) = trainer.initialize_loader()
+        batch = next(enumerate(train_loader))[1]
+
+        assert 2 == 2
```
Comment (williamluer) [inline tests/tests/models/test_tacotron2.py:29]: Remove this

File: tests/tests/models/test_tacotron2.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from uberduck_ml_dev.models.tacotron2 import DEFAULTS as TACOTRON2_DEFAULTS
+from uberduck_ml_dev.models.tacotron2 import Tacotron2
+from uberduck_ml_dev.trainer.tacotron2 import Tacotron2Trainer
+import json
+from uberduck_ml_dev.vendor.tfcompat.hparam import HParams
+import torch
+
+
+class TestTacotron2Model:
+    def test_tacotron2_model(self):
```
Comment (williamluer) [inline tests/tests/models/test_tacotron2.py:10]: Does this test do anything? I don't see any asserts

File: tests/tests/models/test_tacotron2.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            sampler,
+            collate_fn,
+        ) = trainer.initialize_loader()
+        batch = next(enumerate(train_loader))[1]
+
+        assert 2 == 2
+
+        # X, y = model.parse_batch(batch)
+        # forward_output = model(X)
+        # assert len(forward_output) == 4
```
Comment (williamluer) [inline tests/tests/models/test_tacotron2.py:33]: Remove this

File: tests/tests/test_data_loader.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            8000,
+            1024,
+            256,
+            padding=None,
+            win_length=1024,
+            debug=True,
+            debug_dataset_size=12,
+            symbol_set="default",
+        )
+        len(ds)
```
Comment (williamluer) [inline tests/tests/test_data_loader.py:38]: Remove

File: tests/tests/test_data_loader.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        assert len(ds) == 1
+        collate_fn = TextMelCollate(include_f0=True)
+        dl = DataLoader(ds, 12, collate_fn=collate_fn)
+        for i, batch in enumerate(dl):
+            # text_padded,
+            # input_lengths,
+            # mel_padded,
+            # gate_padded,
+            # output_lengths,
+            # speaker_ids,
```
Comment (williamluer) [inline tests/tests/test_data_loader.py:72]: Remove

File: tests/tests/test_data_loader.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        )
+        assert len(ds) == 1
+        collate_fn = TextMelCollate(n_frames_per_step=5, include_f0=True)
+        dl = DataLoader(ds, 12, collate_fn=collate_fn)
+        # text_padded,
+        # input_lengths,
+        # mel_padded,
+        # gate_padded,
+        # output_lengths,
+        # speaker_ids,
```
Comment (williamluer) [inline tests/tests/test_data_loader.py:116]: Remove

File: tests/tests/text/test_symbols.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        assert (
+            len(
+                arpabet_to_sequence(
+                    "{ S IY } { EH M } { Y UW } { D IH K SH AH N EH R IY }"
+                )
+            )
+            == 15
+        )
+        assert arpabet_to_sequence("{ S IY }") == [168, 148]
+        # But symbols_to_sequence hanldes whitespace
```
Comment (williamluer) [inline tests/tests/text/test_symbols.py:17]: Maybe remove?

File: tests/tests/text/test_util.py [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+        assert (
+            len(
+                text_to_sequence(
+                    "Not bad bart, not bad at all.", ["english_cleaners"], p_arpabet=1.0
+                )
+            )
+            == 28
+        )
+        seq = text_to_sequence(
```
Comment (williamluer) [inline tests/tests/text/test_util.py:44]: I think you want to turn these into assert statements

File: tests/tests/trainer/test_trainer.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+            symbol_set="default",
+        )
+        trainer = TTSTrainer(hp)
+        assert trainer.hparams == hp
+        assert trainer.foo == "bar"
+        assert trainer.baz == 123
+        mel = torch.load("tests/fixtures/stevejobs-1.pt")
+        audio = trainer.sample(mel)
+        assert audio.size(0) == 1
+        # trainer.save_checkpoint("test", foo="bar", baz="blah")
```
Comment (williamluer) [inline tests/tests/trainer/test_trainer.py:24]: Remove

File: tests/tests/test_data_loader.py (hunk 5 above)
Comment (sjkoelle) [reply tests/tests/test_data_loader.py:72]: this and the next should be kept for clarity until we have named batched classes - otherwise too confusing

File: tests/tests/test_data_loader.py (hunk 6 above)
Comment (sjkoelle) [reply tests/tests/test_data_loader.py:116]: same

File: tests/tests/models/test_tacotron2.py (hunk 2 above)
Comment (sjkoelle) [reply tests/tests/models/test_tacotron2.py:10]: oops yeah it was the commented out one not the 2=2

File: tests/tests/models/test_tacotron2.py (hunk 1 above)
Comment (sjkoelle) [reply tests/tests/models/test_tacotron2.py:29]: yep (see below)

**Comment by williamluer** [comment]
Nice job moving everything over to pytest. Just left some comments around cleanup and making sure tests have the asserts

File: tests/tests/text/test_util.py (hunk 8 above)
Comment (sjkoelle) [reply tests/tests/text/test_util.py:44]: ehh i dunno - didnt write this - will remove.

File: tests/tests/models/test_tacotron2.py (hunk 3 above)
Comment (sjkoelle) [reply tests/tests/models/test_tacotron2.py:33]: thats the real test :)

File: tests/tests/test_data_loader.py (hunk 5 above)
Comment (sjkoelle) [reply tests/tests/test_data_loader.py:72]: hmmm not sure why in test tho - will remove

File: tests/tests/test_data_loader.py (hunk 4 above)
Comment (sjkoelle) [reply tests/tests/test_data_loader.py:38]: replacing with assert

**Comment by sjkoelle** [comment]
Yeah I think we should just keep the LJ files - useful to have a real batch if we ever wanted to say drive loss to 0.

File: tests/tests/text/test_symbols.py (hunk 7 above)
Comment (sjkoelle) [reply tests/tests/text/test_symbols.py:17]: i dont really know what these comments mean but they look meaningful / intentional

**Comment by sjkoelle** [comment]
K think everything is responded to - some minor changes to the asserts and clean up

**Review by williamluer — APPROVED** [review]
